Sources/Metaplex/Modules/NFTS/Operations/CreateNftOnChainOperationHandler.swift

```diff
@@ -0,0 +1,106 @@
+
+class CreateNftOnChainOperationHandler: OperationHandler {
+    var metaplex: Metaplex
+
+    typealias I = CreateNftInput
+    typealias O = NFT
+
+    init(metaplex: Metaplex) {
+        self.metaplex = metaplex
+    }
+
+    func handle(operation: CreateNftOperation) -> OperationResult<NFT, OperationError> {
+        let builder = InstructionBuilder(metaplex: metaplex)
+        return operation.flatMap { input in
+            OperationResult<[TransactionInstruction], Error>.init { callback in
+                builder.createNftInstructions(input: input) {
+                    callback($0)
+                }
+            }.mapError {
+                OperationError.buildInstructionsError($0)
+            }.flatMap { instructions in
+                OperationResult<String, Error>.init { callback in
+                    self.metaplex.connection.serializeTransaction(instructions: instructions, recentBlockhash: nil, signers: [input.account, input.mintAccountState.account]) {
+                        callback($0)
+                    }
+                }
+                .mapError { OperationError.serializeTransactionError($0) }
+            }.flatMap { serializedTransaction in
+                OperationResult<TransactionID, IdentityDriverError>.init { callback in
+                    self.metaplex.sendTransaction(serializedTransaction: serializedTransaction) {
+                        callback($0)
+                    }
+                }
+                .mapError { OperationError.sendTransactionError($0) }
+            }.flatMap { signature in
+                let operation: () -> OperationResult<SignatureStatus, Retry<Error>> = {
+                    OperationResult<SignatureStatus, Error>.init { callback in
+                        sleep(3)
+                        self.metaplex.confirmTransaction(
+                            signature: signature,
+                            configs: nil
+                        ) { result in
+                            switch result {
+                            case .success(let signature):
+                                guard let signature = signature, let status = signature.confirmationStatus, status == .finalized else {
+                                    callback(.failure(OperationError.nilSignatureStatus))
+                                    return
+                                }
+                                callback(.success(signature))
+                            case .failure(let error):
+                                callback(.failure(error))
+                            }
+                        }
+                    }
+                    .mapError { Retry.retry($0) }
+                }
+                let retry = OperationResult<SignatureStatus, Error>.retry(attempts: 5, operation: operation)
+                    .mapError { OperationError.confirmTransactionError($0) }
+                return retry
```

Comment:
I really don't like this. But I tried various different ways to use `commitment.finalized`

Some things I tried:
Adding `searchTransactionHistory` to `RequestConfiguration`
Passing `searchTransactionHistory` and `Commitment.finalized` to `RequestConfiguration`
Doing one of the other

None appeared to work, so may want you to look at it to see if you have any luck.